internal/rdap/rdap.go

```diff
@@ -44,17 +45,29 @@ var (
 type rdapClient struct{}
 
 // NewClient returns a new RDAP client.
-func NewClient() client.Client {
+func NewClient() rdapClient {
 	return rdapClient{}
 }
 
-func (rdapClient) ExpireTime(domain string) (time.Time, error) {
+func (rdapClient) ExpireTime(ctx context.Context, domain string) (time.Time, error) {
 	log.Debug().Msgf("trying rdap client for %s", domain)
+	req := &rdap.Request{
+		Type:  rdap.DomainRequest,
+		Query: domain,
+	}
+	req = req.WithContext(ctx)
+
 	client := &rdap.Client{}
-	body, err := client.QueryDomain(domain)
+	resp, err := client.Do(req)
 	if err != nil {
-		return time.Now(), err
+		return time.Now(), errors.Wrap(err, "failed to do rdap request")
```

Comment:
```suggestion
		return time.Now(), fmt.Errorf("failed to do rdap request: %w", err)
```

don't need to use pkg/errors